tests/test_cli/test_cli.py

```diff
@@ -1,46 +1,151 @@
 """Tests all command-line subcommands."""
 
+from collections.abc import Generator
+from pathlib import Path
+from typing import Optional
+
 import pytest
 import yaml
-from click.testing import CliRunner
+from click.testing import CliRunner, Result
 from linkml_runtime import SchemaView
-from linkml_runtime.linkml_model import SchemaDefinition
 
 from linkml_map.cli.cli import main
 from tests import (
     DENORM_SPECIFICATION,
     FLATTENING_DATA,
     NORM_SCHEMA,
     PERSONINFO_CONTAINER_DATA,
+    PERSONINFO_DERIVED,
     PERSONINFO_SRC_SCHEMA,
     PERSONINFO_TR,
+    PERSONINFO_TR_COMPILED_MD,
+    PERSONINFO_TR_COMPILED_PY,
 )
 
+DERIVED_SCHEMA_NAME_LINE = "name: personinfo-derived"
+
 
 @pytest.fixture
 def runner() -> CliRunner:
+    """
+    Command line interface test runner.
+
+    :return: command line interface runner
+    :rtype: CliRunner
+    """
     return CliRunner(mix_stderr=False)
 
 
 def test_main_help(runner: CliRunner) -> None:
+    """
+    Ensure that the help command contains the appropriate text.
+
+    :param runner: command line interface runner
+    :type runner: CliRunner
+    """
     result = runner.invoke(main, ["--help"])
     out = result.stdout
     assert "derive-schema" in out
     assert "map-data" in out
     assert result.exit_code == 0
 
 
-def test_derive_schema(runner: CliRunner) -> None:
+def check_result(result: Result, expected_file: Path, output_param: Optional[str] = None) -> None:
+    """
+    Check that the result of running a function matches the expected output.
+
+    :param result: result object from the CliRunner
+    :type result: Result
+    :param expected_file: path to the expected result
+    :type expected_file: Path
+    :param output_param: output param supplied to the function, defaults to None
+    :type output_param: Optional[str], optional
+    """
+    if output_param:
+        with output_param.open() as fh:
+            function_output = fh.read()
+    else:
+        function_output = result.stdout
+
+    with expected_file.open() as fh:
+        assert fh.read() == function_output
+
+
+@pytest.mark.parametrize("output", [None, "output.py"])
+@pytest.mark.parametrize("target", [None, "python", "markdown", "klingon"])
+def test_compile(
```

Comment:
Add in test for the CLI compiler, parametrised by whether or not an output file is provided, and different language inputs.